paimon-core/src/main/java/org/apache/paimon/AbstractFileStore.java

```diff
@@ -360,6 +360,8 @@ public TagDeletion newTagDeletion() {
 
     public abstract Comparator<InternalRow> newKeyComparator();
 
+    protected abstract List<CommitCallback> createCommitCallbacks(String commitUser);
```

Comment:
Can you move `AbstractFileStoreTable.createCommitCallbacks` to here and remove `newCommit(user, callbacks)` method?